src/Resources/Payment.php

```diff
@@ -851,15 +851,15 @@ private function updateChargeTransaction($transaction): void
     private function updateReversalTransaction($transaction): void
     {
         $transactionId = IdService::getResourceIdFromUrl($transaction->url, IdStrings::CANCEL);
-        $authorization = $this->getAuthorization(true);
-        if (!$authorization instanceof Authorization) {
+        $initialTransaction = $this->getInitialTransaction(true);
+        if (!$initialTransaction instanceof Authorization && !$initialTransaction instanceof Charge) {
             throw new RuntimeException('The Authorization object can not be found.');
```

Comment:
Maybe update the error message

Reply:
Good point, I updated it. Please have another look.